src-tauri/src/telemetry_manager.rs

```diff
@@ -371,18 +376,35 @@ async fn get_telemetry_data(
         (false, false) => TelemetryResource::None,
     };
 
-    let p2pool_gpu_stats_sha3 = p2pool_stats.get("sha3").map(|stats| stats.squad.clone());
-    let p2pool_cpu_stats_randomx = p2pool_stats.get("randomx").map(|stats| stats.squad.clone());
+    let p2pool_gpu_stats_sha3 = p2pool_stats.as_ref().map(|s| s.sha3x_stats.clone());
+    let p2pool_cpu_stats_randomx = p2pool_stats.as_ref().map(|s| s.randomx_stats.clone());
     let p2pool_enabled =
         config_guard.p2pool_enabled() && p2pool_manager.is_running().await.unwrap_or(false);
-    let cpu_tribe_name = p2pool_cpu_stats_randomx
-        .clone()
-        .map(|tribe| tribe.name.clone());
-    let cpu_tribe_id = p2pool_cpu_stats_randomx.map(|tribe| tribe.id.clone());
-    let gpu_tribe_name = p2pool_gpu_stats_sha3
-        .clone()
-        .map(|tribe| tribe.name.clone());
-    let gpu_tribe_id = p2pool_gpu_stats_sha3.map(|tribe| tribe.id.clone());
+    let (cpu_tribe_name, cpu_tribe_id) = if p2pool_enabled {
+        if let Some(randomx_stats) = p2pool_cpu_stats_randomx {
+            (
+                Some(randomx_stats.squad.name.clone()),
+                Some(randomx_stats.squad.id.clone()),
+            )
+        } else {
+            (None, None)
+        }
+    } else {
+        (None, None)
+    };
+
+    let (gpu_tribe_name, gpu_tribe_id) = if p2pool_enabled {
+        if let Some(sha3_stats) = p2pool_gpu_stats_sha3 {
+            (
+                Some(sha3_stats.squad.name.clone()),
+                Some(sha3_stats.squad.id.clone()),
+            )
+        } else {
+            (None, None)
+        }
+    } else {
+        (None, None)
+    };
```

Comment:
This section is my biggest gripe, but I don't have a suggestion to fix so 🤷🏻‍♂️

Reply:
yeah hacky.